src/osd/OSD.cc

```diff
@@ -8934,14 +8934,16 @@ void OSD::ShardedOpWQ::_process(uint32_t thread_index, heartbeat_handle_d *hb )
         reqid.name._num, reqid.tid, reqid.inc);
   }
 
-  lgeneric_subdout(osd->cct, osd, 30) << "dequeue status: ";
-  Formatter *f = Formatter::create("json");
-  f->open_object_section("q");
-  dump(f);
-  f->close_section();
-  f->flush(*_dout);
-  delete f;
-  *_dout << dendl;
+  if (g_conf->osd_dump_op_queue) {
+    lgeneric_subdout(osd->cct, osd, 30) << "dequeue status: ";
```

Comment:
This isn't necessary.  The dout macros are a conditional on the debug level and open a set of braces that are later closed by dendl.  Anything between the initial dout (or variant) and dendl macros is conditional.